extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/CredentialConfig.java

```diff
@@ -44,22 +40,19 @@ public class CredentialConfig {
      * If the database is specified in neither place, the default value is {@code admin}. This option is only
      * respected when using the MONGO-CR mechanism (the default).
      */
-    @ConfigItem
-    public Optional<String> authSource;
+    Optional<String> authSource();
 
     /**
      * Allows passing authentication mechanism properties.
      */
-    @ConfigItem
     @ConfigDocMapKey("property-key")
-    public Map<String, String> authMechanismProperties;
+    Map<String, String> authMechanismProperties();
 
     /**
      * The credentials provider name
      */
-    @ConfigItem
-    @ConvertWith(TrimmedStringConverter.class)
-    public Optional<String> credentialsProvider = Optional.empty();
+    @WithConverter(TrimmedStringConverter.class)
+    Optional<String> credentialsProvider();
```

Comment:
We want `Optional<@WithConverter(TrimmedStringConverter.class) String>` and same for the other occurrences.